File: js_project/Strings/FirsthalfSlicer.js
let s1 = "Abhishek";
let s2 = "";
let length = s1.length;
// In JS, division can result in decimals if the length is odd. 
// Math.floor() ensures we get a clean whole number just like Java's integer division.
let midpoint = Math.floor(length / 2);

for (let i = 0; i < midpoint; i++) {
    s2 = s2 + s1.charAt(i);
}

console.log("Original String: " + s1);
console.log("Total Length   : " + length);
console.log("Midpoint Index : " + midpoint);
console.log("First Half Only: " + s2);

File: java_project/Strings/FirsthalfSlicer.java
package Strings;

public class FirsthalfSlicer {
    public static void main(String[] args) {
        String s1 = "Abhishek";
        String s2 = "";
        int length = s1.length();
        int midpoint = length / 2;

        for (int i =0; i<midpoint ; i++){
            s2 = s2+s1.charAt(i);
        }
        System.out.println("Original String: " + s1);
        System.out.println("Total Length   : " + length);
        System.out.println("Midpoint Index : " + midpoint);
        System.out.println("First Half Only: " + s2);
    }
}
